### PR TITLE
Use gcp project id, not name

### LINKED ISSUE #4: pass optional filepath to InitializeBuildRun
this will make it easier to use this util as a library, rather than looking at the env for the config file location

(default to env if no value is passed to function)

### LINKED ISSUE #5: use project id, not project name, in gcr
project id is what google cloud uses in it's APIs
project id != project name (except for rare cases)

### PR DESCRIPTION
**Pull request by mitchdraft**
this is a breaking change but only squash cares right now.
Happy to bump minor version if you like
BOT NOTES: 
resolves https://github.com/solo-io/build/issues/5
resolves https://github.com/solo-io/build/issues/4

### CODE REVIEW

File: pkg/ingest/setup_test.go [hunk 1]
```diff
@@ -3,11 +3,94 @@ package ingest
...
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("could not parse build spec: open solo-project.yaml: no such file or directory"))
+			Expect(br).To(Equal(v1.BuildRun{}))
+		})
+		It("should read from file when provided", func() {
+			br, err := InitializeBuildRun(relativePathToSampleConfig, explicitBuildEnvVars)
+			Expect(err).ToNot(HaveOccurred())
+
+			// aside on matching large structs
+			// Gomega has MatchAllFields, but it requires you to fill in everything, including the XXXabcd fields made by protoc
```
Comment (rickducott) [inline pkg/ingest/setup_test.go]: You can use `proto.Equal`

File: pkg/ingest/setup_test.go [hunk 2]
```diff
@@ -3,11 +3,94 @@ package ingest
...
+			// An alternative is to pass IgnoreExtras with MatchFields
+			// Either way, you have to write out a lot of stuff by hand
+			// This is worthwhile in some cases because you can be explicit about how fields should match and you get
+			// great, scoped feedback on errors
```
Comment (rickducott) [inline pkg/ingest/setup_test.go]: Agree this gives scoped errors on field differences, but at the expense of:
- test readability
- ignoring extras, which could allow future bugs

File: pkg/ingest/setup_test.go [hunk 3]
```diff
@@ -3,11 +3,94 @@ package ingest
...
+						})),
+					})),
+				})),
+			}))
+			// It is much easier to do this - stringify the struct
+			// One shortcoming is that you only get +/- 5 chars of context
+			// per: https://github.com/onsi/gomega/blob/master/format/format.go#L146
+			// TODO(mitchdraft) gomega pr to modify charactersAroundMismatchToInclude (if not merged will make a util)
+			Expect(br.String()).To(Equal(sampleConfigIngested.String()))
+			// This was my first thought, but the error message is not human friendly, you have to compare large blocks of text
```
Comment (rickducott) [inline pkg/ingest/setup_test.go]: Is there a utility that could be written here for better diffing in the event of a failed expect (similar to your string context case)? This is the optimal test for readability

File: pkg/ingest/setup_test.go [hunk 4]
```diff
@@ -3,11 +3,94 @@ package ingest
...
 var _ = Describe("ingest config", func() {
 
+	const relativePathToSampleConfig = "../../examples/sample-solo-project.yaml"
+	var buildId = "some-build-id"
+	const nonReleaseTagValue = ""
+	var explicitBuildEnvVars = &v1.BuildEnvVars{
+		BuildId:       buildId,
+		TaggedVersion: nonReleaseTagValue,
+	}
+	var sampleConfigIngested = v1.BuildRun{
```
Comment (rickducott) [inline pkg/ingest/setup_test.go]: nit: this isn't the ingested config, would refer to this as "expectedConfig" or something similar

File: pkg/ingest/setup_test.go (hunk 1 above)
Comment (mitchdraft) [reply pkg/ingest/setup_test.go]: ooo, nice, I'll use that in the util

File: pkg/ingest/setup_test.go (hunk 4 above)
Comment (mitchdraft) [reply pkg/ingest/setup_test.go]: fixed

File: pkg/ingest/setup_test.go (hunk 3 above)
Comment (mitchdraft) [reply pkg/ingest/setup_test.go]: let me know what you think of ExpectEqualProtoMessages, I can move it to go-utils if it looks good

File: pkg/ingest/setup_test.go [hunk 5]
```diff
@@ -4,10 +4,66 @@ import (
...
+			ComputedBuildVars: &v1.ComputedBuildVars{
+				Release:         false,
+				ImageTag:        buildId,
+				ContainerPrefix: "gcr.io/solo-public-1010",
+				Version:         buildId,
+			},
+		},
+	}
+
+	FContext("InitializeBuildRun", func() {
```
Comment (rickducott) [inline pkg/ingest/setup_test.go]: unfocus before merge

**Review by rickducott — APPROVED** [review]

**Pull request merged**